### PR TITLE
 Separated cases with balance due in summary

### PR DESCRIPTION
**Pull request by wittejm**
* Add empty data field for cases/charges with balance
 * Exclude charges of cases with balance in existing summary fields

Addresses #1530

### CODE REVIEW

**Comment by wittejm** [comment]
Terrible approach ^^ I think maybe we expand the organizer code here: 
https://github.com/codeforpdx/recordexpungPDX/blob/f25ca3a8aa4f5e1acbdcd75d11decc871a110f60/src/backend/expungeservice/record_summarizer.py#L51

And the summarized charges structure is extended from 
`Dict[str, List[Tuple[str, str]]]`
to
`Dict[str, List[Tuple[str,List[Tuple[str, str]]]]]`
So that charges are always grouped under their case numbers. 
Then the frontend render shows the case number only in the sections that have balance and omits it in others.

File: src/backend/expungeservice/record_summarizer.py [hunk 1]
```diff
@@ -74,28 +84,50 @@ def secondary_sort(charge: Charge):
                 return date.max()
 
         def get_label(charge: Charge):
+            no_balance = RecordSummarizer._get_case_by_case_number(record, charge.case_number).summary.balance_due_in_cents == 0
             charge_eligibility = charge.expungement_result.charge_eligibility
             if charge_eligibility:
-                return charge_eligibility.label
+                if charge_eligibility.label != "Needs More Analysis" and charge_eligibility.label != "Ineligible" and  not no_balance:
```
Comment (wittejm) [inline src/backend/expungeservice/record_summarizer.py:90]: modus tollens this

Comment (wittejm) [reply src/backend/expungeservice/record_summarizer.py:90]: No. de Morgans

File: src/backend/expungeservice/record_summarizer.py [hunk 2]
```diff
@@ -74,28 +84,50 @@ def secondary_sort(charge: Charge):
...
+            charges_under_label : List[Tuple[str,List[Tuple[str, str]]]]= []
+            for case_number, case_charges in groupby(charges, key=lambda charge: charge.case_number):
+                case = RecordSummarizer._get_case_by_case_number(record, case_number)
+                case_info_line = RecordSummarizer._get_case_balance_header_info_for_case(case)
+                charges_tuples = [
+                    (case_charge.ambiguous_charge_id, case_charge.to_one_line())
+                    for case_charge in case_charges
+                    if case_charge.edit_status != EditStatus.DELETE
+                ]
+                charges_under_label.append((case_info_line, charges_tuples))
```
Comment (wittejm) [inline src/backend/expungeservice/record_summarizer.py:114]: rename to `charges_in_section`

File: src/backend/expungeservice/record_summarizer.py [hunk 3]
```diff
@@ -74,28 +84,50 @@ def secondary_sort(charge: Charge):
...
             else:
                 return ""  # TODO: Rethink if possible
 
         SHOW_ALL_CHARGES_THRESHOLD = 20
         if len(record.charges) <= SHOW_ALL_CHARGES_THRESHOLD:
             visible_charges = record.charges
         else:
-            visible_charges = [charge for charge in record.charges if not charge.charge_type.hidden_in_record_summary()]
-        eligible_charges_by_date: Dict[str, List[Tuple[str, str]]] = {}
+            visible_charges = [charge for charge in record.charges_without_case_balance if not charge.charge_type.hidden_in_record_summary()]
```
Comment (wittejm) [inline src/backend/expungeservice/record_summarizer.py:101]: undo this change

File: src/frontend/src/components/RecordSearch/Record/RecordSummary/ChargesList.tsx [hunk 4]
```diff
@@ -11,34 +11,15 @@ export default class ChargesList extends React.Component<Props> {
...
-        eligibilityDate === "Eligible Now"
-          ? "green"
-          : eligibilityDate === "Ineligible"
-          ? "red"
-          : eligibilityDate === "Needs More Analysis"
-          ? "purple"
-          : "dark-blue";
-      const SHOW_ALL_CHARGES_THRESHOLD = 20;
+    ).map(([eligibilityDate, casesWithHeaderAndChargesNames]: [string, any[]]) => {
+      console.log(casesWithHeaderAndChargesNames);
```
Comment (wittejm) [inline src/frontend/src/components/RecordSearch/Record/RecordSummary/ChargesList.tsx:15]: ya got me

**Comment by wittejm** [comment]
The new formatting for charges that have a balance on their case looks like this: The new line showing case number, county, and balance is in bold and the charges for that case/balance are indented. Charges that don't have a balance aren't indented. 

with_case_balance_header
 
@hmarcks please LMK if you have suggestions on the tweaked formatting.

I also removed the red "$" badge on charges because the info is redundant with these new "if balance paid" sections. 

Ready for code review @KentShikama

**Comment by wittejm** [comment]
Oh I suppose these Codacy notes are actually useful. Reviewers stand by...

**Review by KentShikama — APPROVED** [review]

**Comment by hmarcks** [comment]
Sorry I'm late on this @wittejm , one thing that confused me a bit was that the marijuana case is listed in Eligible Now and in Eligible Now If Balance Paid. Would if clear things up if it was only listed under Eligible Now If Balance Paid?

**Comment by wittejm** [comment]
That example has two different marijuana violation charges on the record and only one of them has a balance.

**Comment by hmarcks** [comment]
Got it, the same date had me confused but that wouldn’t be likely
